src/main/java/com/cloudant/search3/SearchServer.java

```diff
@@ -0,0 +1,273 @@
+import io.netty.handler.codec.http.HttpServerKeepAliveHandler;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.ssl.SslContext;
+import io.netty.util.concurrent.DefaultEventExecutorGroup;
+import io.netty.util.concurrent.EventExecutorGroup;
+
+public class SearchServer extends AbstractServer {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final String RPC_MESSAGE_TYPE = "rpc-message-type";
+
+    private static class SearchServerHandler extends SimpleChannelInboundHandler<HttpObject> {
+
+        private static final ByteBuf EMPTY = wrappedBuffer(Empty.getDefaultInstance().toByteArray());
+
+        private final Search search;
+
+        public SearchServerHandler(final Search search) {
+            this.search = search;
+        }
+
+        @Override
+        public void channelReadComplete(final ChannelHandlerContext ctx) {
+            ctx.flush();
+        }
+
+        @Override
+        protected void channelRead0(final ChannelHandlerContext ctx, final HttpObject msg) throws Exception {
+            if (msg instanceof FullHttpRequest) {
+                final FullHttpRequest req = (FullHttpRequest) msg;
+
+                if (!POST.equals(req.method())) {
+                    ctx.writeAndFlush(error(Type.CLIENT_ERROR, req.uri() + " method not allowed"));
+                    return;
+                }
+
+                final String messageType;
+                final ByteBuf result;
+                switch (req.uri()) {
+                case "/Search/Delete": {
+                    final Index index = decode(Index.getDefaultInstance(), req);
+                    search.delete(index);
+                    messageType = "empty";
+                    result = EMPTY;
+                    break;
+                }
+                case "/Search/Info": {
+                    final Index index = decode(Index.getDefaultInstance(), req);
+                    messageType = "info_response";
+                    result = encode(search.info(index));
+                    break;
+                }
+                case "/Search/Search": {
+                    final SearchRequest request = decode(SearchRequest.getDefaultInstance(), req);
+                    messageType = "search_response";
+                    result = encode(search.search(request));
+                    break;
+                }
+                case "/Search/GroupSearch": {
+                    final GroupSearchRequest request = decode(GroupSearchRequest.getDefaultInstance(), req);
+                    messageType = "group_search_response";
+                    result = encode(search.groupSearch(request));
+                    break;
+                }
+                case "/Search/UpdateDocument": {
+                    final DocumentUpdateRequest request = decode(DocumentUpdateRequest.getDefaultInstance(), req);
+                    messageType = "session_response";
```

Comment:
I just want to verify you intended `session_response` as the message type for update, delete, set update sequence?

Reply:
Yes, that's what the protocol says (in the `service` stanza that I deleted as part of this PR as it no longer has an effect on the generated code). All a session response means is "yes, I did your operation and I'm still the session you expect." the alternative is an error response.